src/Data/Aeson/QQ.hs

```diff
@@ -32,6 +41,14 @@ jsonExp txt =
   where
     parsed' = QQ.parsedJson txt
 
+
+jsonFileExp :: String -> ExpQ
+jsonFileExp fileName = do
+  addDependentFile fileName
+  txt <- runIO $ readFile fileName
+  jsonExp txt
```

Comment:
@koterpillar: Do you know about https://hackage.haskell.org/package/template-haskell-2.11.1.0/docs/Language-Haskell-TH-Quote.html#v:quoteFile ?